File: old/2023/rs/01.rs
fn part_one(inp: &Vec<Vec<char>>) -> u32 {
    inp.iter()
        .map(|chs| {
            (*chs.iter().find(|c| c.is_digit(10)).unwrap() as u8 - b'0') as u32 * 10
                + (*chs.iter().rev().find(|c| c.is_digit(10)).unwrap() as u8 - b'0') as u32
        })
    .sum::<u32>()
}

fn main() {
    let inp = std::fs::read_to_string("../input/01").unwrap().trim().to_string();
    let inp = inp.lines().map(|c| c.chars().collect::<Vec<_>>()).collect::<Vec<_>>();

    println!("Part one: {}", part_one(&inp));
    //println!("Part two: {}", part_two(&inp));
}

File: 2024/d01-historian_hysteria/langs/01.hs
module Main where

parse :: [String] -> [[Int]]
parse [x]
parse (x:xs) = 

main :: IO ()
main = do 
    inp <- readFile "../../../input/2024/01"
    putStrLn ("Part one: " ++ (show (part_one inp 0)))
    putStrLn ("Part two: " ++ (show (part_two inp 0 0)))
